src/ui/layout/mct-tree.vue

```diff
@@ -959,14 +953,33 @@ export default {
         handleTreeResize() {
             this.calculateHeights();
         },
-        destroyObservers(observers) {
-            Object.entries(observers).forEach(([keyString, unobserve]) => {
-                if (typeof unobserve === 'function') {
-                    unobserve();
-                }
+        /**
+         * Destroy the given observer and its children.
+         * If no navigationPath is provided, destroys all observers.
+         */
+        destroyObservers(navigationPath = '') {
+            Object.entries(this.observers).filter(([key]) => key.includes(navigationPath))
```

Comment:
I like the idea of destroying all if no nav path is passed in, but can we short circuit if a path is passed in and just destroy that one? There could be cases where thousands of tree items are in the tree, we could just skip past filtering all of them. Same for mutables below.

Reply:
Yep, this was horribly inefficient. I was able to piggyback on the logic used in `closeTreeItemByPath` and destroy specific observers/mutables while we're filtering the tree.